src/linematch.c

```diff
@@ -0,0 +1,402 @@
+#include <assert.h>
+#include <math.h>
+#include <stdbool.h>
```

Comment:
can we get rid of that and use ints instead?

Reply:
converted from bool to int